pkgs/top-level/perl-packages.nix

```diff
@@ -1221,6 +1221,23 @@ let
     };
   };
 
+  BarcodeZBar = buildPerlPackage {
+    pname = "Barcode-ZBar";
+    version = "0.04";
+    src = "${zbar.src}/perl";
```

Comment:
What's the difference with https://metacpan.org/release/SPADIX/Barcode-ZBar-0.04 ? Some glance at https://github.com/mchehab/zbar/commits/master/perl tells me there's substantial changes, but users may be confused as to the provenance of this module.

Might be worth convincing upstream to push a new version to CPAN.

Reply:
The metacpan module was last updated in 2009 and uses a zbar version from that time (before it was forked onto github).

The changes in the github repo are to update the module so that it works with the current version.

Unfortunately it doesn't seem like the cpan maintainer is around anymore, not sure how cpan modules work exactly but could the current maintainer of zbar take ownership of that? Or would it be better to suggest that he makes a new cpan module (similar to graphviz and graphviz2)?

Reply:
Yes, there is a process for adopting CPAN modules outlined in http://neilb.org/2013/07/24/adopt-a-module.html that upstream can do - I'd be happy to lend a hand and talk to the CPAN admins if needed :muscle: 

It is also possible to make a new `Barcode::ZBar2` dist like what GraphViz did - the namespace changes will be necessary due to how CPAN arranges ownership of namespaces, if not taking the adoption route.

Reply:
So Barcode-ZBar 0.04 on CPAN is not the same as `${zbar.src}/perl` ?
 
`NetServerSSPrefork` was a similar case where the we went for an updated version from the authors repo, instead of the latest on CPAN. I suggest marking the version like "0.04pre" or "unstable-YYYY-MM-DD" if the CPAN version can't be used.

Reply:
That is correct the `zbar.src` is different from the CPAN version. The Perl module was forked at the same time that the main project was and has been updated since to maintain compatability.

Reply:
Ive made an issue on the ZBar repo

Reply:
In the meantime we could mark this version with the version of ZBar (0.23.90) that it supports?

Reply:
Works for me. 

Can you add a short comment describing the reason for it? 

Also, I'd prefer setting something like `version = "0.04pre"` since it doesn't deviate from `$Barcode::ZBar::VERSION` - if that's ok.

Reply:
Sure ill do that now

Reply:
Just a little nitpick: Can you move the comment to i.e. above the src attribute, so it is more clear that it belongs to this derivation? :)

Reply:
Any idea why the tests are failing? Its building on my machine with
`nix build .#perlPackages.BarcodeZBar`

Reply:
The packages build fine for me too, however the basic evaluation checks fail with:
```
...
2021-08-25T14:05:55.8834116Z checking Nixpkgs on aarch64-darwin
2021-08-25T14:06:35.4812678Z error: while querying the derivation named 'perl5.32.1-Barcode-ZBar-0.04pre':
2021-08-25T14:06:35.4815914Z while evaluating the attribute '__impureHostDeps' of the derivation 'perl5.32.1-Barcode-ZBar-0.04pre' at /nix/store/jzigajkia0hjyjq1xakyfkcpisxqfdfm-source/pkgs/stdenv/generic/make-derivation.nix:205:7:
2021-08-25T14:06:35.4829281Z while evaluating anonymous function at /nix/store/jzigajkia0hjyjq1xakyfkcpisxqfdfm-source/pkgs/stdenv/generic/make-derivation.nix:188:32, called from /nix/store/jzigajkia0hjyjq1xakyfkcpisxqfdfm-source/pkgs/stdenv/generic/make-derivation.nix:188:17:
2021-08-25T14:06:35.4832733Z while evaluating 'chooseDevOutputs' at /nix/store/jzigajkia0hjyjq1xakyfkcpisxqfdfm-source/lib/attrsets.nix:500:22, called from undefined position:
2021-08-25T14:06:35.4835556Z undefined variable 'TestMore' at /nix/store/jzigajkia0hjyjq1xakyfkcpisxqfdfm-source/pkgs/top-level/perl-packages.nix:1234:20
2021-08-25T14:06:35.5536880Z builder for '/nix/store/6cy9172l569fpkylir23939kianysr7j-nixpkgs-release-checks.drv' failed with exit code 1
2021-08-25T14:06:35.5735589Z error: build of '/nix/store/6cy9172l569fpkylir23939kianysr7j-nixpkgs-release-checks.drv' failed
2021-08-25T14:06:35.5783136Z ##[error]Process completed with exit code 100.
```
https://github.com/NixOS/nixpkgs/pull/135667/checks?check_run_id=3422793076

I'm not sure why it fails, but could be an issue with the action or the platform it's running on.